### DISCUSSION TITLE
How to create scenarios dinamically?

### DISCUSSION THREAD

**Discussion by stephannv** [lookbook-hq/lookbook#583] [Q&A] [answered]
Hello,

I'm trying to define some scenarios programatically, eg.
```ruby
# @!group Colors
ButtonComponent::COLORS.each do |color|
  define_method :"with_#{color}_color" do
    render ButtonComponent.new(color: color) do
      "Create product"
    end
  end
end
# @!endgroup
```

But it isn't working, these scenarios don't appear on Lookbook. Am I missing something? Or Lookbook doesn't accept metaprogramming for scenarios?

**Comment by mryurii** [comment, marked as answer]
It's not supported – https://github.com/ViewComponent/lookbook/issues/154#issuecomment-1253866122

Seems like one `#with_all_colors` scenario that renders multiple component variations  is the way to go

**Reply by Jackson200** [reply]
Thanks very much @mryurii.
Really appreciate the pointer. Missed that one.
